app/services/stripe/cancel_subscription_service.rb

```diff
@@ -0,0 +1,18 @@
+module Stripe
+  class CancelSubscriptionService
+    def initialize(subscription)
+      @subscription = subscription
+    end
+
+    def call
+      ActiveRecord::Base.transaction do
+        @subscription.cancel_stripe_subscription && @subscription.cancel_local_subscription
```

Comment:
need to cancel on local first.

also I don't think `cancel_local_subscription` will raise in exception always if it fails?? which means if it fails it will fail silently.

There is no point of wrapping in transaction if failing logic doesn't raise exception.

Reply:
manual testing is not reliable, we cannot say i tested and it was not breaking. got to cancel method and see whats going on, what if it fails what it will return etc, look at docs and find proof there, answer me with solid documentation and proof rather than manual testing which might have missed many cases.

Reply:
verified it with you